app/models/cart.rb

```diff
@@ -1,5 +1,7 @@
 class Cart < ApplicationRecord
   has_many :line_items, dependent: :destroy
+  has_many :products, through: :line_items
+  has_many :enabled_products, -> {where enabled: true}, through: :line_items, source: :product
```

Comment:
create scope for `where enabled: true ` in products.
Also use parentheses around where for better readability